### ISSUE TITLE
AMD graphics card not supported?

### ISSUE THREAD

**Issue by Sauler** [Equilibrium-Games/Acid#36]
I wanted to try out Acid in my project. But when i'm trying to run one of demo projects I got error:
`Expression: false && "Vulkan runtime error, failed to find a suitable gpu!"`

I debugged code and find out the **VK_KHR_push_descriptor** vulkan extension is not supported on AMD GPU. I found [thread ](https://community.amd.com/thread/233640)on AMD forum. There is suggestion to use **VK_KHR_descriptor_update_template** instead.

System specs:
- **System:** Windows 10 Pro 1809 x64
- **CPU:** AMD Ryzen 7 2700X
- **RAM:** 32GB
- **GPU:** AMD Radeon RX 570 Series

**Comment by Sauler** [comment]
Now I got Unknown vulkan error -1000069000. From debbuger I can see that error is in this line:
![image](https://user-images.githubusercontent.com/5783815/51075491-6c512300-168c-11e9-9f72-e1c29e512940.png)
_DescriptiorSet.cpp:24_

**Comment by mattparks** [comment]
What test are you running, and when your not running with the debugger what is the console full output? Also what does the call history look like before it hits the error.

**Comment by Sauler** [comment]
I'm running **TestGUI**, but this error is in all of them.
Full console output: https://pastebin.com/9XrRDrq0
Full call history:
![image](https://user-images.githubusercontent.com/5783815/51089504-a4c92d80-176e-11e9-84e7-69fdd5e1b873.png)

In Renderer::CheckVk method result is set to **VK_ERROR_OUT_OF_POOL_MEMORY**.

**Comment by mattparks** [comment]
There was an error that I fixed in 75b70b1392c49c4fbacdeb9bfdcad9eaf9439cb1 that would not allocate enough pool space in pipeline descriptor pools for the types used in the shader. I was not having any pool memory issues before or after making those changes so I cannot be sure if it worked; https://github.com/Equilibrium-Games/Acid/blob/master/Sources/Renderer/Pipelines/PipelineGraphics.cpp#L182 might be an issue if pool creation is too large.

**Comment by Sauler** [comment]
It still does not work. Same error in same line.

**Comment by mattparks** [comment]
Can you tell me what driver you are using, also update to the latest driver. Have you tried to use any other Vulkan engines/demos? You might want to change the value in PipelineGraphics.cpp on line 182 (linked in my last comment) to a value under 128, 64 will still run for me, I am thinking the pool is not working correctly on AMD.

You might also want to debug what this line of code is providing to the descriptor pool: https://github.com/Equilibrium-Games/Acid/blob/master/Sources/Renderer/Pipelines/PipelineGraphics.cpp#L177
I am getting a list of pools that looks like this (from Font/Gui):
![image](https://user-images.githubusercontent.com/4019937/51091485-cddac580-1750-11e9-9472-f463f1a9336a.png)

**Comment by Sauler** [comment]
I have newest driver. Vulkan examples that are shipped with SDK works fine.
Changing maxSets value does nothing.
I am getting same list as you.

**Comment by mattparks** [comment]
I was able to upgrade from my NVidia 960 to an AMD R9 390. For some reason on NVidia pipeline pool sizes did not matter when allocating, so I set each descriptor pool size to the max for each type in the pipeline. This was a quick hack to get it running, oddly has no memory/speed impact with having such a large count. I am also seeing some texture artifacts in Physics/PBR, something I am trying to fix now.
